packages/shared-components/src/components/TimeScrubber/TimeScrubber.js

```diff
@@ -0,0 +1,375 @@
+  }
+
+  /**
+   * Restricts a position to only be within the allowed range
+   */
+  capToRange = position => {
+    const { tickSpacing, allowedRange } = this.props;
+    const rangeSpan = (allowedRange[1] - allowedRange[0]) * 4 * tickSpacing;
+    const minimum = allowedRange[0] * 4 * tickSpacing;
+    const maximum = minimum + rangeSpan;
+    return Math.min(maximum, Math.max(minimum, position));
+  };
+
+  /**
+   * Takes the current time value and rounds it to 15 minutes.
+   * Called whenever a user-provided value is supplied to the component
+   * to ensure it is always rounded.
+   */
+  enforceTimeRounding = () => {
+    const { value, onChange } = this.props;
+    if (onChange && value.minutes() % 15 !== 0) {
+      const rounded = value
+        .clone()
+        .minutes(roundMinutes(value.minutes()))
+        .seconds(0);
+      onChange(rounded);
+    }
+  };
+
+  /**
+   * Returns the physical timeline position based on the current
+   * scrubbed time. Optionally, you can provide your own base time value
+   * as the first parameter. In both cases, any offset from user interaction
+   * will be added to the final position.
+   *
+   * You can pass a time object to override the internal value.
+   */
+  getPosition = timeValue => {
+    const { tickSpacing } = this.props;
+    const { internalValue } = this.state;
+    const { offset } = this;
+    const value = timeValue || internalValue;
+    const hours = toHours(value);
+    const rawPosition = tickSpacing * hours * 4 - offset;
+
+    return this.capToRange(rawPosition);
+  };
+
+  /**
+   * Converts the current real scrubber position into a time value.
+   */
+  getTime = timeValue => {
+    const { tickSpacing } = this.props;
+    const { internalValue } = this.state;
+    const realPosition = this.getPosition(timeValue);
+    const hoursWithPartial = realPosition / 4 / tickSpacing;
+    const hours = Math.floor(hoursWithPartial);
+    const minutes = Math.floor((hoursWithPartial - hours) * 60);
+    const roundedMinutes = roundMinutes(minutes);
+    return (
+      timeValue ||
+      internalValue
+        .clone()
+        .hours(hours)
+        .minutes(roundedMinutes)
+    );
+  };
+
+  /**
+   * Bumps the time up or down by the provided number of minutes.
+   */
+  stepTime = stepMinutes => {
+    const { onChange, allowedRange } = this.props;
+    const { internalValue } = this.state;
+    const newValue = internalValue.clone().add(stepMinutes, 'minutes');
+    const hoursWithPartial = toHours(newValue);
+    if (
+      hoursWithPartial < allowedRange[0] ||
+      hoursWithPartial > allowedRange[1]
+    ) {
+      return;
+    }
+    this.setState({ internalValue: newValue });
+    if (onChange) {
+      onChange(newValue);
+    }
+  };
+  incrementTime = () => this.stepTime(15);
+  decrementTime = () => this.stepTime(-15);
+  canIncrementTime = () =>
+    toHours(this.state.internalValue) < this.props.allowedRange[1];
```

Comment:
This assumes that the allowedRange provided is always even with the step value, which isn't the safest of assumptions.